File: packages/sdui/src/main/java/nl/q42/core/Locale.java
package nl.q42.core;

public enum Locale
{
  NL_NL("nl-NL"),
  EN_US("en-US"),
  EN_UK("en-UK");

  public final String value;

  Locale(String value)
  {
    this.value = value;
  }

  public static Locale from(String value, Locale defaultLocale)
  {
    for (Locale locale : Locale.values())
    {
      if (locale.value.equals(value))
        return locale;
    }

    return defaultLocale;
  }
}

File: packages/sdui/src/main/kotlin/nl/q42/core/Locale.kt
package nl.q42.core

enum class Locale
    (val value: String) {
    NL_NL("nl-NL"),
    EN_US("en-US"),
    EN_UK("en-UK");

    companion object {
        fun from(value: String?, defaultLocale: Locale?): Locale? {
            for (locale in entries) {
                if (locale.value == value) return locale
            }

            return defaultLocale
        }
    }
}
